src/Hydrator/HydratorInterface.php

```diff
@@ -9,11 +9,11 @@
 
 namespace Zend\Stdlib\Hydrator;
 
-use Zend\Stdlib\Extractor\ExtractionInterface;
+use Zend\Hydrator\HydratorInterface as BaseHydratorInterface;
 
 /**
  * @deprecated Use Zend\Hydrator\HydratorInterface from zendframework/zend-hydrator instead.
  */
-interface HydratorInterface extends HydrationInterface, ExtractionInterface
+interface HydratorInterface extends BaseHydratorInterface
```

Comment:
I think BaseHydratorInterface should be append to the extend list and not replace previous interfaces.

This IMO the way of maintain compatibility is appending new interfaces and maintain the old ones in the list too.